### PR TITLE
Separate quality check and docker build

### PR DESCRIPTION
**Pull request by roynrishingha**
All tests are done in release mode with `--features=runtime-benchmarks` enabled.
Caching is enabled to speed up workflow. It caches `.cargo` and `target` directories.

    - `develop-pr` action is dedicated to check code quality on `develop` branch on every PR submit.
    - `release-docker-image` action is dedicated for releasing docker image.
      This action has nothing to do with checking or testing the code.

### CODE REVIEW

**Comment by roynrishingha** [comment]
@amarts while building the image we are tagging it with `dev-0.8.0`. Should it be tagged with `develop` ?

**Comment by amarts** [comment]
> @amarts building the image we are tag it with `dev-0.8.0`. Should it be tagged with `develop` ?

Can we use the branch name? so that it doesn't matter which branch we push for in future, it gets the corresponding tag. For now, `develop` should be used.

File: .github/workflows/cord-ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  pull_request:
+    branches: [ develop, dev-0.8.0]
+    paths-ignore:
+      - "doc/**"
+      - "**.md"
+  push:
+    branches: [ develop, dev-0.8.0]
+    paths-ignore:
+      - "doc/**"
+      - "**.md"
```
Comment (amarts) [inline .github/workflows/cord-ci.yml]: can we separate out the `push` to a separate file? that way, we know which one runs. And do we need a same test to run again?

Comment (roynrishingha) [reply .github/workflows/cord-ci.yml]: Done.

File: .github/workflows/dev-pr.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            ~/.cargo/
+            target/
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Check code syntax
+        run: cargo check --release --all --all-targets --locked --verbose --color always
+
+      - name: Run tests
+        run: cargo test --release --all --all-targets --locked --features=runtime-benchmarks --no-fail-fast --verbose --color always
```
Comment (amarts) [inline .github/workflows/dev-pr.yml]: looking at `develop-pr.yml`, i feel this is not required. Remove this file.

File: .github/workflows/dev-pr.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+name: pr-submit
+on:
+  pull_request:
+    branches: [ dev-0.8.0]
```
Comment (amarts) [inline .github/workflows/dev-pr.yml]: this will 'never' run in 'develop' branch. (and this commit is for develop branch).

**Comment by roynrishingha** [comment]
Removed `dev-pr.yml`

**Review by amarts — APPROVED** [review]

**Pull request merged**